test/e2e/authorizer/serviceaccounts_test.go

```diff
@@ -0,0 +1,199 @@
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"configmaps"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+		},
+	}, metav1.CreateOptions{})
+	require.NoError(t, err, "failed to create role")
+
+	t.Log("Creating role binding to access configmaps")
+	_, err = kubeClient.RbacV1().RoleBindings(namespace.Name).Create(ctx, &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "sa-access-configmap",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      "default",
+				Namespace: namespace.Name,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "Role",
+			Name:     "sa-access-configmap",
+			APIGroup: rbacv1.GroupName,
+		},
+	}, metav1.CreateOptions{})
+	require.NoError(t, err, "failed to create role")
+
+	t.Log("Waiting for service account to be created")
+	require.Eventually(t, func() bool {
+		_, err := kubeClient.CoreV1().ServiceAccounts(namespace.Name).Get(ctx, "default", metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return false
+		} else if err != nil {
+			t.Fatalf("unexpected error retrieving service account: %v", err)
+		}
+		return true
+	}, wait.ForeverTestTimeout, time.Millisecond*100, "\"default\" service account not created in namespace %s",
+		helper.QualifiedObjectName(namespace),
+	)
+
+	t.Log("Waiting for service account secret to be created")
+	var tokenSecret corev1.Secret
+	require.Eventually(t, func() bool {
+		secrets, err := kubeClient.CoreV1().Secrets(namespace.Name).List(ctx, metav1.ListOptions{})
+		require.NoError(t, err, "failed to list secrets")
+
+		for _, secret := range secrets.Items {
+			if secret.Annotations[corev1.ServiceAccountNameKey] == "default" {
+				tokenSecret = secret
+				return true
+			}
+		}
+		return false
+	}, wait.ForeverTestTimeout, time.Millisecond*100, "token secret for default service account not created")
+
+	type TokenTestCase struct {
+		name  string
+		token func(t *testing.T) string
+	}
+	for _, ttc := range []TokenTestCase{
+		{"Legacy token", func(t *testing.T) string {
+			return string(tokenSecret.Data["token"])
+		}},
+		{"Bound service token", func(t *testing.T) string {
+			t.Log("Creating service account bound token")
+			boundToken, err := kubeClient.CoreV1().ServiceAccounts(namespace.Name).CreateToken(ctx, "default", &authenticationv1.TokenRequest{
+				Spec: authenticationv1.TokenRequestSpec{
+					Audiences:         []string{"https://kcp.default.svc"},
+					ExpirationSeconds: pointer.Int64Ptr(3600),
+					BoundObjectRef: &authenticationv1.BoundObjectReference{
+						APIVersion: "v1",
+						Kind:       "Secret",
+						Name:       tokenSecret.Name,
+						UID:        tokenSecret.UID,
+					},
+				},
+			}, metav1.CreateOptions{})
+			require.NoError(t, err, "failed to create token")
+			return boundToken.Status.Token
+		}},
+	} {
+		t.Run(ttc.name, func(t *testing.T) {
+			saRestConfig, err := server.DefaultConfig()
```

Comment:
(No action required) I'm going to update DefaultConfig to accept `t` so it can use require on any errors. Test helpers that require checking of trivial errors is an anti-pattern.